pyphare/pyphare/pharesee/geometry.py

```diff
@@ -165,19 +238,52 @@ def get_periodic_list(patches, domain_box):
     of the sorted list, if that patch touches the upper (resp. lower)
     border of the domain
     """
+    assert len(patches) > 0
+    dim = patches[-1].box.ndim
+    assert all([p.box.ndim == dim for p in patches])
+
     from copy import copy
 
-    sorted_patches = sorted(patches, key=lambda p: p.origin)
+    sorted_patches = sorted(patches, key=lambda p: p.origin.all())
 
-    if touch_domain_border(sorted_patches[-1].box, domain_box, "upper"):
-        last_patch = copy(sorted_patches[-1])
-        shift_patch(last_patch, -domain_box.upper)
-        sorted_patches.insert(0, last_patch)
+    if dim == 1: # only two possible border patches, [0] and [-1]
+        if touch_domain_border(sorted_patches[-1].box, domain_box, "upper"):
+            last_patch = copy(sorted_patches[-1])
+            shift_patch(last_patch, -domain_box.upper)
+            sorted_patches.insert(0, last_patch)
 
-    if touch_domain_border(sorted_patches[0].box, domain_box, "lower"):
-        first_patch = copy(sorted_patches[0])
-        shift_patch(first_patch, domain_box.upper)
-        sorted_patches.append(first_patch)
+        if touch_domain_border(sorted_patches[0].box, domain_box, "lower"):
+            first_patch = copy(sorted_patches[0])
+            shift_patch(first_patch, domain_box.upper)
+            sorted_patches.append(first_patch)
+
+    if dim == 2:
+        for patch in patches: # not sure sorting can help 2d
+            box = patch.box
+
+            left = box.lower[0] == domain_box.lower[0]
```

Comment:
I think this is wrong. I think it's if the ghostbox has a chance to breach the boundary, we want to copy/shift periodically. this is just an adjacency check, so insufficient.

Reply:
Additions to the overlap tests should help

Reply:
#377 handles the 1d changes this is expected to be similar to